src/widget/tree_structure.rs

```diff
@@ -62,15 +62,32 @@ impl TreeStructure {
             .or_insert(parent_id);
     }
 
-    /// # Panics
-    ///
-    /// When the `parent_id` doesn't exist in the structure or `range` is out of bounds this will panic
-    pub(crate) fn delete_children(&mut self, parent_id: Id, range: std::ops::Range<usize>) {
-        let children = &self.children[&parent_id][range.clone()];
-        for child in children {
-            self.parent.remove(child);
+    /// Deletes all children of `parent_id`, but not their descendants
+    pub(crate) fn delete_children(&mut self, parent_id: Id, range: impl RangeBounds<usize>) {
```

Comment:
This is not needed anymore and I'm not sure if there's a case where this is relevant anymore (it's slightly more efficient than `delete_descendants` as it doesn't require an allocation, but this is only relevant if there are no further descendants, otherwise it could be a pitfall)

Should I rename `delete_descendants` to `delete_children` then (when `delete_children` is removed)?

Reply:
Answering my own question: I have removed that for now, as it's well not used, and I don't see an immediate need for it (and CI also wants it to be removed...)